pkg/apis/pipeline/v1alpha1/task_defaults.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2018 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+func (t *Task) SetDefaults() {
+	t.Spec.SetDefaults()
+}
+
+func (ts *TaskSpec) SetDefaults() {
+	ts.SetDefaults()
```

Comment:
This is going to infinitely recurse, eventually opening a wormhole that will swallow the earth. Or, stack overflow, whichever is first.

Reply:
![image](https://user-images.githubusercontent.com/432502/46506159-404e1800-c7e8-11e8-822c-459f3c50c742.png)